packages/block-library/src/shortcode/edit.js

```diff
@@ -10,8 +10,8 @@ const ShortcodeEdit = ( { attributes, setAttributes, instanceId } ) => {
 	const inputId = `blocks-shortcode-input-${ instanceId }`;
 
 	return (
-		<div className="wp-block-shortcode">
-			<label htmlFor={ inputId }>
+		<div className="wp-block-shortcode  components-placeholder">
```

Comment:
@jasmussen - can you advice how it could be coded differently? In general, we assume that class names are local to the component which define them. In this case, it's `wp.components.Placeholder` which uses `components-placeholder`. While it seems to be very convenient to use an existing class, it doesn't scale well as we learned down the road.

Reply:
Yeah this is an interesting question.

I would say there are two ways to go. 

1. Keep the shortcode block as its own thing. But then we remove the placeholder classes and _duplicate_ the CSS rules necessary for them to look the same.
2. Rewrite/potentially redesign the shortcode to actually _use_ the placeholder component. 

2 is a bigger task, and I have no strong opinions on which one is the right way to go. What do you think?

Reply:
What is a placeholder component? What does it mean? Does it make sense as a separate component?

Reply:
One of the main issues with duplicating the class like this is maintainability. If someone wants to refactor the placeholder styles, they'd generally expect only the `<Placeholder>` component to be affected, but in this case other components can be affected in an unexpected way.

> What is a placeholder component?

In the context of a block, it seems like it's the UI that's displayed when the proper wysiwyg part of a block can't be shown. Outside of the context of a block, I'm not really sure. The word 'placeholder' generally implies a stopgap that will eventually be replaced, but I'm not sure how that applies to other user interfaces that `@wordpress/components` might be used in.

A couple of other options might be:
1) Extend `Placeholder` to support other layouts, like the horizontal layout shown here
2) Extract the container of the placeholder into a separate component (`PlaceholderCard`?) and use that.